jsonnet/components/prometheus-adapter.libsonnet

```diff
@@ -201,17 +201,17 @@ function(params)
           - cluster:
               certificate-authority: %s
               server: %s
-            name: prometheus-k8s
+            name: thanos-querier
           contexts:
           - context:
-              cluster: prometheus-k8s
-              user: prometheus-k8s
-            name: prometheus-k8s
-          current-context: prometheus-k8s
+              cluster: thanos-querier
+              user: thanos-querier
+            name: thanos-querier
+          current-context: thanos-querier
           kind: Config
           preferences: {}
           users:
-          - name: prometheus-k8s
+          - name: thanos-querier
```

Comment:
I guess it is needed in order to be authorized by thanos-querier kube-rbac-proxy. But I am not a fan of the approach we are currently using to authenticate prometheus-adapter to its backend. The config we are providing is kubeconfig like and is very confusing. I think it would be better if we were to use the more conventional authn/authz parameters: https://github.com/kubernetes-sigs/prometheus-adapter/blob/3206c65b47e968761f8df81f2d545a1e432a0acf/cmd/adapter/adapter.go#L130-L137